Lib/test/test_inspect/test_inspect.py

```diff
@@ -1846,57 +1937,238 @@ class Test:
 class TestIsDataDescriptor(unittest.TestCase):
 
     def test_custom_descriptors(self):
-        class NonDataDescriptor:
-            def __get__(self, value, type=None): pass
-        class DataDescriptor0:
-            def __set__(self, name, value): pass
-        class DataDescriptor1:
-            def __delete__(self, name): pass
-        class DataDescriptor2:
-            __set__ = None
-        self.assertFalse(inspect.isdatadescriptor(NonDataDescriptor()),
-                         'class with only __get__ not a data descriptor')
-        self.assertTrue(inspect.isdatadescriptor(DataDescriptor0()),
-                        'class with __set__ is a data descriptor')
-        self.assertTrue(inspect.isdatadescriptor(DataDescriptor1()),
-                        'class with __delete__ is a data descriptor')
-        self.assertTrue(inspect.isdatadescriptor(DataDescriptor2()),
-                        'class with __set__ = None is a data descriptor')
-
-    def test_slot(self):
-        class Slotted:
-            __slots__ = 'foo',
-        self.assertTrue(inspect.isdatadescriptor(Slotted.foo),
-                        'a slot is a data descriptor')
+        # Custom method descriptors (non-data descriptors):
 
-    def test_property(self):
-        class Propertied:
-            @property
-            def a_property(self):
-                pass
-        self.assertTrue(inspect.isdatadescriptor(Propertied.a_property),
-                        'a property is a data descriptor')
+        class MethodDescriptor:
+            """with __get__ and no __set__/__delete__"""
+            def __get__(self, *_): pass
+        class MethodDescriptorWithGetNone:
+            """with __get__=None and no __set__/__delete__"""
+            __get__ = None
+        class MethodDescriptorSub(MethodDescriptor):
+            """with __get__ (inherited) and no __set__/__delete__"""
+
+        for cls in [
+            MethodDescriptor,
+            MethodDescriptorWithGetNone,
+            MethodDescriptorSub,
+        ]:
+            with self.subTest(f"a descriptor {cls.__doc__}; "
+                              f"its class: {cls.__name__}"):
+                self.assertFalse(inspect.isdatadescriptor(cls()))
 
-    def test_functions(self):
-        class Test(object):
+        # Custom data descriptors:
+
+        class DataDescriptorWithSetOnly:
+            """with __set__ (and no __get__)"""
+            def __set__(self, *_): pass
```

Comment:
Now that I look at it again, I'm really considering to generate the classes using `type()` but I'm not sure it's better.